components/clientComponents/forms/AddressComplete/AddressComplete.tsx

```diff
@@ -0,0 +1,300 @@
+
+    const selectedResult = addressResultCache.find(
+      (item: AddressCompleteChoice) => toFullAddress(item) === value
+    );
+    if (selectedResult === undefined) {
+      return; // Do nothing, this is not found in the AddressComplete API.
+    } else {
+      const responseData = await getSelectedAddress(
+        apiKey,
+        selectedResult.Id,
+        addressObject?.country || "CAN",
+        i18n.language as Language
+      );
+      if (responseData) {
+        const results = responseData;
+        setAddressObject(results);
+        if (comboboxRef.current) {
+          comboboxRef.current.changeInputValue(results.streetAddress);
+        }
+      }
+    }
+  };
+
+  const setAddressData = (key: string, value: string) => {
+    let baseAddressObject = {};
+
+    if (addressObject === null) {
+      baseAddressObject = {
+        streetAddress: "",
+        city: "",
+        province: "",
+        postalCode: "",
+        country: "",
+      };
+    } else {
+      baseAddressObject = addressObject;
+    }
+
+    for (const internalKey in baseAddressObject) {
+      if (key === internalKey) {
+        const newAddressObject = { ...baseAddressObject, [key]: value };
+        setAddressObject(newAddressObject as AddressElements);
+      }
+    }
+  };
+
+  const countryChoices = countries.all?.map((country) => {
+    return country[i18n.language as Language];
+  });
+
+  const setCountry = (countryText: string) => {
+    // Get the country from the countries.all object.
+    const country = countries.all.find(
+      (country) => country[i18n.language as Language] === countryText
+    );
+    if (country) {
+      // set the country in the address object to the ID of the country.
+      setAddressData("country", country.id);
+    }
+  };
+
+  return (
+    <>
+      <fieldset
+        role="group"
+        aria-describedby={ariaDescribedBy ? `desc-${id}` : undefined}
+        data-testid="addressComplete"
+        id={id}
+        tabIndex={0}
+      >
+        {ariaDescribedBy && (
+          <Description id={`desc-${id}`} className="gc-form-group-context">
+            {ariaDescribedBy}
+          </Description>
+        )}
+
+        {props.canadianOnly && (
+          <div>
+            <input type="hidden" id={`${name}-country`} name={`${name}-country`} value={"CAN"} />
+          </div>
+        )}
+        {!props.canadianOnly && (
+          <div className="mb-6">
+            <Label htmlFor={`${name}-country`} className="gc-label">
+              {t("addElementDialog.addressComplete.country")}
+            </Label>
+            <Combobox
+              id={`${name}-country`}
+              name={`${name}-country`}
+              choices={countryChoices}
+              onSetValue={(val) => setCountry(val)}
+              className={cn(meta.error && "gc-error-input")}
+              required={required}
+              defaultValue="Canada"
+              data-testid="addresscomplete-input-country"
+            />
```

Comment:
Wondering why you chose to use Combobox here instead of the ManagedCombobox that you customized? 

I figured the purpose of ManagedCombobox was so you could customize it for your purposes and leave Combobox alone for now, but it looks like you've also made changes to Combobox.

Reply:
The ManagedCombobox was only to prevent issues around the choices being propagated at runtime, and avoid overcomplicating and adding functionality to the Combobox that broke cypress tests.

I could probably swap both to use Managed.

Reply:
Ideally now that we know the managed state works we update the current combobox so we only need the one and not both to maintain.

Would suggest make a separate PR to update the current and pull it into this one now that it's been proved out.

Reply:
@dsamojlenko perhaps in the same PR we could do the ID update.